src/install/lockfile.zig

```diff
@@ -3288,6 +3421,21 @@ pub const Serializer = struct {
             }
         }
 
+        {
+            const slice = lockfile.packages.slice();
+            const name_hashes = slice.items(.name_hash);
+            const resolutions = slice.items(.resolution);
+
+            for (resolutions) |resolution, i| {
```

Comment:
This is a little expensive of a loop to add on every call. I wonder if we could store a workspace_count at the end of the lockfile and only do this loop if the number is greater than 0

Reply:
Merged with the package ID setup loop in https://github.com/oven-sh/bun/pull/1764/commits/3539db0ac230adbc27dcc3fa2d8a28d5c1250989 &minus; extra loop no longer required at all.